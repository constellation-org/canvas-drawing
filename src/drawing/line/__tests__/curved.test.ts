import CurvedLine from '../curved'

describe('[CurvedLine]', () => {
  let canvas: HTMLCanvasElement
  let context: CanvasRenderingContext2D
  beforeEach(function () {
    canvas = document.createElement('canvas')
    canvas.setAttribute('width', '100')
    canvas.setAttribute('height', '100')
    context = canvas.getContext('2d') as CanvasRenderingContext2D
  })
  it('should draw a curved line', () => {
    const line = new CurvedLine({
      from: {
        x: 10,
        y: 10,
      },
      to: {
        x: 90,
        y: 90,
      },
      control: {
        x: 0,
        y: 100,
      }
    })
    line.draw(context)
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
})
