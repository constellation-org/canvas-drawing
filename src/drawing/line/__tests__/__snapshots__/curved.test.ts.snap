// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CurvedLine] should draw a curved line 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAABtJJREFUeF7tnXeo7UQQxn/Pgr33rqigomJFURT9QxF7BbGBXVQUG1YQe8WCHX0qFlRUUOxiQRQRBSsqKth7773xSRaH6H2bk5O5N/fsDISEe09mZ78vm83uzsxOIaRXCEzplTVhDEFIzx6CICQI6RkCPTMnWkgQ0jMEemZOtJAgpGcI9MycJi1kdeBN4Jue2T6S5uQImQ34vqq5CHkZWH8kkehJpXKELA28ZWx9D1iyJ7aPpBk5QtYCnjE1fxZYcySR6EmlcoRsBtxnbH0A0N9CnBDIEbIbcL0p+wZgdydbQi1kJxcPA84zSF0A6G8hTgjkWshpwHGm7OOB051sCbUNWsgVwH4GKV1fGcj5IZBrIbcD25vitwPu8DMnNOcIeQzY0MC0AfBEwOaHQI6QV4AVTfErAK/5mROac4S8DSxlYFoCeD9g80MgR8i7gEhIshjwoZ85oTlHiFqDSEiyCPBxwOaHQI6QD4BFTfELA5/4mROac4R8BIiEJAsCnwVsfgjkCNHraSFT/PzAF37mhOYcIZ8CCxiY5gO+DNj8EMgR8jkgEpLMA3ztZ05ozhGi19O8Bqa5Y23d96HJEfIVIBKSzAl852tS2dpzhGgQqLFHEnXw6ldCnBDIEfIisIopeyXgVSdbQm2D9ZBHgY0MUpr5fTyQ80Mg10JuBXY0xcd6iB8X/2jOEXI5sL+xYV/gKmebilafI+RUQOvoSY4FziwaMefK5wipe52cCxzlbFPR6nOEyAfrOoPQtcCeRSPmXPkcIZsD9xgb7gK2drapaPU5QtYBnjIIvQSsWjRizpXPEaJ5LDvd/icwI6BziAMCOUJUZH2RauUqTsTBnFDZhBB5vG9qoNoZuCWg80GgCSHnAEea4uXve4KPOaG1CSH1kIQ7gW0DOh8EmhCir6oXTPEKAF3Wx5zQ2oSQ6YDfAJ2TzAT8GvB1j0ATQlSqWogdf2wCPNS9OaGxKSGXAAcauM4HDg/4ukegKSH1KRR5wMsTPqRjBJoSoj5DiQN0ThKe8B2TIXVNCdFvNcmolpLkYECvspAOERiEkIOAi03Z9wJbdGhLqBqwhSwOKLVGkl+AuQCdQzpCYJAWoiKVfEauQEn0CrOZHjoyq1w1gxJSn9e6CdilXPi6r/mghKh1qJUk+R1YJuIOuyNmUEJU8iPAxsaEs4BjujOpbE1tCNGaumZ8k8j5WlFWP5YNZTe1b0OI7nmjNuOrT+JLuzGpbC1tCBFihwLKDJREU/LLAX+VDefwtW9LyKxV8KfOSfYCrhnepLI1tCVEqKmFqKUkUfjb8hHyNtwDNQwh6shfB+YwJiid0wHDmVT23cMQIuQOAS40EKoPkXOdTZxZNsID1n5YQqYHnqtFWSnqarXo4Adkovr5sIRIjVLJPl2bqNTq4mXtTCr7ri4IEYL1VIA/AOsBai0hAyDQFSEKndZgUak3kigth/LGR/agCSBERSpu5Opa2Woh6wI/DWBT0T/tqoUkEG8DdqghejewTXjMN3vOuiZEThBKkqmO3oqSMR/RzKSyf9U1IUJT/YjGIdpZwYrGLBeVDXe+9h6EqFRlMhUp2n/EytnA0Xmzyv2FFyFCVO6mWm/X4NGKkhHsWvkLl4v8GDX3JERFagZ46v+UrfQcW0Wqp/8i402IStQX1s3AzLXi36kcJJ6MZvIvAuNBiErTV5dC42wyNP1dwaNy3NYODBHeMKCj3LAPsoJ8Hq5lyk46lfJJ/YomKouW8WohCWRNsSgzhPqPuvxRrTieWHL27PEmJJGwR+UnbBe30v80zSIf4jMApRgsSiaKEIGsJP/6BF57DMSV/VTjFi0VFzMXNpGEiAeNUTRQPAmYYQxilCf4xmpnH6X2GGmZaEISuHIhEilKSmCDS+vgax9FzSiLoJHMH9wXQhLw8h0+udpmaVq2KSpYk5j3V8fILIT1jZBEjNbkTwG2bPh+0iLYg4CCiJS9SIPOSSl9JcS+yvauNrO0+5jkwFa/83w1rtHYRocCVfVp3WvpOyEWPCXA2QfYaQhEFQGmTWq0L4o9lDBaX3I/m3O6Tpm8Z6+mfzQFVD/kwamE09rOQ4cSTqdrnZV8eo0mdk8mQlJ9NKWvcAjtyatjMqT5+LYK/8tyMhkJqVdK7qsiRi1Ic2Z2A5osAOP4A7UweeNMU0aBkHoFtaWGXg9KBZIOfb3VZ5tz2HT9/0Zp2keRkLGAVMvRflp6vyuiWNc69O4XibOM0UekVU8FJKlfUdSxzvZaT76czdVX6NC2UPXrRuOmkgjp+ol30ReEuMDaXmkQ0h47lzuDEBdY2ysNQtpj53JnEOICa3ulQUh77Fzu/BvuQ+1lhe5PXQAAAABJRU5ErkJggg=="`;
